pkg/secrets/censor.go

```diff
@@ -0,0 +1,45 @@
+package secrets
+
+import (
+	"sync"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/test-infra/prow/logrusutil"
+)
+
+// DynamicCensor keeps a list of censored secrets that is dynamically updated.
+// Used when the list of secrets to censor is updated during the execution of
+// the program and cannot be determined in advance.  Access to the list of
+// secrets is internally synchronized.
+type DynamicCensor struct {
+	sync.RWMutex
+	secrets sets.String
+}
+
+func NewDynamicCensor() DynamicCensor {
+	return DynamicCensor{
+		secrets: sets.NewString(),
+	}
+}
+
+// AddSecrets adds the content of one or more secrets to the censor list.
+func (c *DynamicCensor) AddSecrets(s ...string) {
+	c.Lock()
+	defer c.Unlock()
+	c.secrets.Insert(s...)
+}
+
+// GetSecrets returns the current list of secrets to censor.
+// The return value should not be modified.
+func (c *DynamicCensor) GetSecrets() sets.String {
+	c.RLock()
+	defer c.RUnlock()
+	return c.secrets
+}
+
+// Formatter creates a new formatter to be used to filter output.
+func (c *DynamicCensor) Formatter(f logrus.Formatter) logrus.Formatter {
```

Comment:
We do not need another implementation of this. Why don't we (simply) just use a `secretutil.Censorer` ? I'll add another call upstream to update instead of overwrite secrets...

Reply:
Or just do this:

```go
type DynamicCensor struct {
	sync.RWMutex
	delegate secretutil.Censorer
	secrets sets.String
}

func (c *DynamicCensor) AddSecrets(s ...string) {
	c.Lock()
	defer c.Unlock()
	c.secrets.Insert(s...)
	c.delegate.Refresh(c.secrets.List()...)
}

func (c *DyanmicCensor) Censor(input *[]byte) {
	c.delegate.Censor(input)
}
```

That's all you need here.